// useragents/index.js
function firefoxUserAgent(os, version) {
  var platformOsCpu = {
      'osx': 'Macintosh; Intel Mac OS X 10.7',
      'windows': 'Windows NT 6.1' //lazily letting this be windows 7 for now
  };
  
  var str = 'Mozilla/5.0 (' + platformOsCpu[os] + '; rv:' + version + '.0) Gecko/' + version + '.0 Firefox/' + version + '.0';
  return str;
}

function chromeUserAgent(os, version) {
  var platformOs = {
    'osx': 'Macintosh; Intel Mac OS X 10_7_5',
    'windows': 'Windows NT 6.1; WOW64' //windows 7, 64bit processor
  };

  return 'Mozilla/5.0 (' + platformOs[os] + ') AppleWebKit/537.11 (KHTML, like Gecko) Chrome/' + version + '.0.1271.91 Safari/537.11';
}

function ieUserAgent(os, version) {
  var trident = {
    '8': 'Trident/4.0',
    '9': 'Trident/5.0',
    '10': 'Trident/6.0'
  };

  var str = 'Mozilla/5.0 (compatible; MSIE ' + version + '.0; Windows NT 6.1; ' + trident[version] + ')';
  return str;
}

module.exports = function userAgent(browser, os, version) {
  return module.exports[browser](os, version);
};

module.exports.firefox = firefoxUserAgent;
module.exports.chrome = chromeUserAgent;
module.exports.ie = ieUserAgent;
